### PR TITLE
dev-libs/libxslt: version bump

### PR DESCRIPTION
**Pull request by stha09**
Closes: https://bugs.gentoo.org/632214
Package-Manager: Portage-2.3.19, Repoman-2.3.6

### CODE REVIEW

**Comment by stha09** [comment]
I have only tested the ebuild on FreeBSD so far, because the glibc-2.26 patch was wrong there too.

File: dev-libs/libxslt/libxslt-1.1.32.ebuild [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+)
+
+MULTILIB_WRAPPED_HEADERS=(
+	/usr/include/libxslt/xsltconfig.h
+)
+
+src_prepare() {
+	default
+
+	DOCS=( AUTHORS ChangeLog FEATURES NEWS README TODO )
```
Comment (monsieurp) [inline dev-libs/libxslt/libxslt-1.1.32.ebuild]: Please move this array outside the `src_prepare()` function.

File: dev-libs/libxslt/libxslt-1.1.32.ebuild [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+	# https://bugzilla.gnome.org/show_bug.cgi?id=684621
+	eapply "${FILESDIR}"/${PN}.m4-${PN}-1.1.26.patch
```
Comment (monsieurp) [inline dev-libs/libxslt/libxslt-1.1.32.ebuild]: Put all patches into the `PATCHES` array outside this function as well.

File: dev-libs/libxslt/libxslt-1.1.32.ebuild [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+			install
+		python_foreach_impl python_optimize
+	fi
+}
+
+multilib_src_install_all() {
+	einstalldocs
+
+	if ! use examples; then
+		rm -rf "${ED}"/usr/share/doc/${PF}/examples
```
Comment (monsieurp) [inline dev-libs/libxslt/libxslt-1.1.32.ebuild]: `rm -r ... || die`

File: dev-libs/libxslt/libxslt-1.1.32.ebuild [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		rm -rf "${ED}"/usr/share/doc/${PF}/python/examples
```
Comment (monsieurp) [inline dev-libs/libxslt/libxslt-1.1.32.ebuild]: Ditto.

**Comment by leio** [comment]
Please don't make the requested changes for a version bump not proxy-maintained. Maybe in a separate commit though. That said:

- rm -rf can never die, and `|| die` is redundant and misleading
- DOCS in global scope or elsewhere is a maintainer style preference and isn't a place to change as non-maintainer. That said, I don't see a reason to not change it - things other than libxslt and libxml2 that we maintain have it in global scope. There is no practical different - both cases end up in saved environment.
- PATCHES array does not support conditional patching, that the prefix people have added, and at that point it is bad to use PATCHES array in global scope, as the order in which the patches there and the conditional one in src_prepare are applied isn't immediately obvious.

IOW, please disregard the review. I was planning to look over it as libxslt co-maintainer within a couple days now and merge it if all is fine after functional review. Some of these suggested changes would be wrong and would make me reject the PR. More useful would be knowing more handily if any actual changes beyond KEYWORDS dropping to ~arch was needed or not, though.. but will see after pulling the patch and diffing with previous version ebuild.

**Comment by monsieurp** [comment]
`rm -rf` in an ebuild is plain wrong. But sure, disregard the review.

**Comment by leio** [comment]
oh, I missed the removal of -f, yeah, that's maybe good change too.

**Comment by stha09** [comment]
So just remove -f from all rm's?

File: dev-libs/libxslt/libxslt-1.1.32.ebuild (hunk 3 above)
Comment (EvaSDK) [reply dev-libs/libxslt/libxslt-1.1.32.ebuild]: rm -rf cannot fail

**Comment by EvaSDK** [comment]
Also, please make sure testsuite runs fine. I'm not sure this was the case last time I tried to bump this package and it is pretty critical.

**Comment by charIes17** [comment]
Please add closing bug 637310 to the commit message.
Closes: https://bugs.gentoo.org/637310

**Comment by leio** [comment]
I forgot about this PR before doing the bump from scratch; though unfortunately I found it afterwards from bugzilla comments, making me waste half an hour on the rm -r stuff, which obviously wasn't as easy as the suggested change, for zero actual benefits.

**Pull request closed without merging**